### PR TITLE
Remove refresh-pairs-cached

### PR DESCRIPTION
**Pull request by xmatthias**
## Summary
As deprecated in #2156, this will remove -r / --refresh-pairs-cached.
Unfortunately not everything depending could be removed since edge still requires to update the data "during edge" in real operations.

This is explained in a TODO and next steps for this are outlined there.

## Quick changelog

- remove deprecated option -r / --refresh-pairs-cached
- clarify download-data incremental option

### CODE REVIEW

File: docs/backtesting.md [hunk 1]
```diff
@@ -37,6 +37,10 @@ This will download ticker data for all the currency pairs you defined in `pairs.
 - Use `--timeframes` to specify which tickers to download. Default is `--timeframes 1m 5m` which will download 1-minute and 5-minute tickers.
 - To use exchange, timeframe and list of pairs as defined in your configuration file, use the `-c/--config` option. With this, the script uses the whitelist defined in the config as the list of currency pairs to download data for and does not require the pairs.json file. You can combine `-c/--config` with most other options.
 
+!!! Tip Updating existing data
+    If you already have backtesting data available in your data-directory and would like to refresh this data up to today, use `--days xx` with a number slightly higher than the missing number of days. Freqtrade will load the available data and only download the missing data.
```
Comment (hroff-1902) [inline docs/backtesting.md]: "will keep the available data"

File: docs/deprecated.md [hunk 2]
```diff
@@ -4,17 +4,15 @@ This page contains description of the command line arguments, configuration para
...
 ### the `--refresh-pairs-cached` command line option
 
 `--refresh-pairs-cached` in the context of backtesting, hyperopt and edge allows to refresh candle data for backtesting.
 Since this leads to much confusion, and slows down backtesting (while not being part of backtesting) this has been singled out 
 as a seperate freqtrade subcommand `freqtrade download-data`.
 
-This command line option was deprecated in `2019.7-dev` and will be removed after the next release.
-
-## Removed features
+This command line option was deprecated in `2019.7-dev` and removed in `2019-9`
```
Comment (hroff-1902) [inline docs/deprecated.md]: "in `2019.9`"

File: freqtrade/data/history.py [hunk 3]
```diff
@@ -142,33 +141,25 @@ def load_data(datadir: Path,
...
-              live: bool = False
               ) -> Dict[str, DataFrame]:
     """
-    Loads ticker history data for a list of pairs the given parameters
+    Loads ticker history data for a list of pairs
     :return: dict(<pair>:<tickerlist>)
+    TODO: refresh_pairs is still used by edge to keep the data uptodate.
+        This should be replaced in the future. Instead, writing the current candles to disk
+        from dataprovider should be implemented, as this would avoid loading ohlcv data twice.
+        exchange and refresh_pairs are then not needed here nor in load_pair_history.
```
Comment (hroff-1902) [inline freqtrade/data/history.py]: (comment, no actions needed): right, I also thought of this reviewing data/history.py...

File: freqtrade/data/history.py [hunk 4]
```diff
@@ -142,33 +141,25 @@ def load_data(datadir: Path,
...
+
+    for pair in pairs:
+        hist = load_pair_history(pair=pair, ticker_interval=ticker_interval,
+                                 datadir=datadir, timerange=timerange,
+                                 refresh_pairs=refresh_pairs,
+                                 exchange=exchange,
+                                 fill_up_missing=fill_up_missing)
+        if hist is not None:
+            result[pair] = hist
     return result
```
Comment (hroff-1902) [inline freqtrade/data/history.py]: Hooray, I waited for this moment for long time (I mean the removal all this 'live' part from this function)...

👏

Comment (xmatthias) [reply freqtrade/data/history.py]: funnily, the --live part could have been removed earlier, since that was removed already ... :D

**Pull request merged**